src/NHibernate/Linq/Visitors/QueryModelVisitor.cs

```diff
@@ -379,6 +391,38 @@ public override void VisitResultOperator(ResultOperatorBase resultOperator, Quer
 			ResultOperatorMap.Process(resultOperator, this, _hqlTree);
 		}
 
+		private bool TryGetRelatedFetchRequest(NhJoinClause joinClause, QueryModel queryModel, out FetchRequestBase fetchRequest)
+		{
+			if (joinClause.Restrictions.Count > 0 ||
+			    !(joinClause.FromExpression is MemberExpression memberExpression) ||
+			    !(memberExpression.Expression is QuerySourceReferenceExpression querySource) ||
+				// Hql does not support fetch with group by and select
+			    queryModel.ResultOperators.Any(o => o is GroupResultOperator) ||
```

Comment:
Can you clarify why this check is needed? Is it some kind of optimization?

Reply:
Reusing a join for group by and select does not work, the fetch join won't be added to the query, so a normal hql join must be used. I didn't add the test for group by as it already exists in `GroupByTests`:
https://github.com/nhibernate/nhibernate-core/blob/aadc6ce23ba659abbd91ba3e6518f799d2a5e4f3/src/NHibernate.Test/Linq/ByMethod/GroupByTests.cs#L905-L910